server/rest/src/main/java/org/infinispan/rest/resources/ServerResource.java

```diff
@@ -332,26 +344,82 @@ private CompletionStage<RestResponse> threads(RestRequest restRequest) {
 
    private CompletionStage<RestResponse> report(RestRequest restRequest) {
       ServerManagement server = invocationHelper.getServer();
-      NettyRestResponse.Builder responseBuilder = new NettyRestResponse.Builder();
       return Security.doAs(restRequest.getSubject(), (PrivilegedAction<CompletionStage<RestResponse>>) () ->
             server.getServerReport().handle((path, t) -> {
                if (t != null) {
-                  return responseBuilder.status(HttpResponseStatus.INTERNAL_SERVER_ERROR).build();
-               } else {
-                  return responseBuilder
-                        .contentType(MediaType.fromString("application/gzip"))
-                        .header("Content-Disposition",
-                              String.format("attachment; filename=\"%s-%s-%3$tY%3$tm%3$td%3$tH%3$tM%3$tS-report.tar.gz\"",
-                                    Version.getBrandName().toLowerCase().replaceAll("\\s", "-"),
-                                    invocationHelper.getRestCacheManager().getNodeName(),
-                                    Calendar.getInstance())
-                        )
-                        .entity(path.toFile()).build();
+                  return new NettyRestResponse.Builder().status(HttpResponseStatus.INTERNAL_SERVER_ERROR).build();
                }
+
+               return createReportResponse(path.toFile(), invocationHelper.getRestCacheManager().getNodeName());
             })
       );
    }
 
+   private RestResponse createReportResponse(Object report, String filename) {
+      return new NettyRestResponse.Builder()
+            .contentType(MediaType.fromString("application/gzip"))
+            .header("Content-Disposition",
+                  String.format("attachment; filename=\"%s-%s-%3$tY%3$tm%3$td%3$tH%3$tM%3$tS-report.tar.gz\"",
+                        Version.getBrandName().toLowerCase().replaceAll("\\s", "-"),
+                        filename,
+                        Calendar.getInstance())
+            )
+            .entity(report)
+            .build();
+   }
+
+   private CompletionStage<RestResponse> nodeReport(RestRequest restRequest) {
+      String targetName = restRequest.variables().get("nodeName");
+      EmbeddedCacheManager cacheManager = invocationHelper.getProtocolServer().getCacheManager();
+      NettyRestResponse.Builder responseBuilder = new NettyRestResponse.Builder();
+      List<Address> members = cacheManager.getMembers();
+
+      if (invocationHelper.getRestCacheManager().getNodeName().equals(targetName)) {
+         return report(restRequest);
+      }
+
+      if (members == null) {
+         return CompletableFuture.completedFuture(responseBuilder.status(NOT_FOUND).build());
+      }
+
+      ByRef<ByteBuffer> response = new ByRef<>(null);
+      return SecurityActions.getClusterExecutor(cacheManager)
+            .submitConsumer(ecm -> {
```

Comment:
Is there no way we can translate the `targetName` to an address and use the `filterTargets` method? Otherwise we are broadcasting this message to all nodes and only one is doing actual work.

Reply:
Yeah, this is the one I told you about. Same problem as the key distribution one since we don't know the other node's name :( That would greatly help both PRs